### ISSUE TITLE
[Bug]: Tokens not created after successful password validation

### ISSUE THREAD

**Issue by sebastianbuechler** [zitadel/zitadel#5267]
### Preflight Checklist

- [X] I could not find a solution in the documentation, the existing issues or discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Environment

ZITADEL Cloud

### Describe the bug

A user reported that sometimes his login does not work on the first trial. In the event logs I can find multiple occasions like this:
![image](https://user-images.githubusercontent.com/42775578/220589286-a433fa77-f432-4954-af33-3bc789a39fc3.png)

### To reproduce

1. Login with password
2. Fail to retrieve access token
3. Try again until it works

**Comment by hifabienne** [comment]
So far we found out that this is a problem with Firefox.
We are investigating more on our side.

**Comment by maxximino** [comment]
I can reproduce the bug, with oauth2-proxy (client configured with PKCE).
With a Firefox incognito window, the first log in with password fails.
The `POST /oauth/v2/token` made by oauth2-proxy gets this response:
```
{
  "error": "invalid_grant",
  "error_description": "invalid code challenge"
}
```
so oauth2-proxy returns 500.
When retrying immediately, zitadel won't ask for login again (just to pick the user), and it does just work.
Works on Chrome incognito window.

**Comment by TakumiHaruta** [comment]
We've encountered the same issue in Chrome incognito window.
It occurs when `/token` endpoint is called after leaving the service's screen for a long time. Refreshing the page does not bring me back to the default URL

**Comment by livio-a** [comment]
The problem of the issue creator was caused by Firefox not redirecting back to the native app. The user needs to approve by clicking on button. In the meantime the browser would retry to open the callback and reopen the prompt to allow the redirect again. 
We therefore introduced (#5627, released with 2.24) an option on native apps, to skip the success page after login:
![image](https://user-images.githubusercontent.com/9405495/233553571-91b53037-6708-4845-b773-cbf124025cc9.png)

**Comment by TakumiHaruta** [comment]
Okay thanks. I'll create a new issue then
